packages/driver/src/cypress/log.ts

```diff
@@ -497,139 +496,142 @@ const Log = function (cy, state, config, obj) {
+      return trigger(log, 'command:log:changed')
     }
 
-    const triggerLog = function (log) {
-      log._hasInitiallyLogged = true
+    // only fire the log:state:changed event
+    // as fast as every 4ms
+    // @ts-ignore
+    log.fireChangeEvent = _.debounce(triggerStateChanged, 4)
 
-      return trigger(log, 'command:log:added')
-    }
+    log.set(options)
 
-    const addToLogs = function (log) {
-      const id = log.get('id')
+    // if snapshot was passed
+    // in, go ahead and snapshot
+    if (log.get('snapshot')) {
+      log.snapshot()
+    }
 
-      logs[id] = true
+    // if end was passed in
+    // go ahead and end
+    if (log.get('end')) {
+      log.end()
     }
 
-    const logFn = function (options = {}) {
-      if (!_.isObject(options)) {
-        $errUtils.throwErrByPath('log.invalid_argument', { args: { arg: options } })
-      }
+    if (log.get('error')) {
+      log.error(log.get('error'))
+    }
 
-      const log = Log(cy, state, config, options)
+    log.wrapConsoleProps()
 
-      // add event emitter interface
-      $Events.extend(log)
+    const onBeforeLog = state('onBeforeLog')
 
-      const triggerStateChanged = () => {
-        return trigger(log, 'command:log:changed')
+    // dont trigger log if this function
+    // explicitly returns false
+    if (_.isFunction(onBeforeLog)) {
+      if (onBeforeLog.call(cy, log) === false) {
+        return
       }
+    }
 
-      // only fire the log:state:changed event
-      // as fast as every 4ms
-      log.fireChangeEvent = _.debounce(triggerStateChanged, 4)
+    // set the log on the command
+    const current = state('current')
 
-      log.set(options)
+    if (current) {
+      current.log(log)
+    }
 
-      // if snapshot was passed
-      // in, go ahead and snapshot
-      if (log.get('snapshot')) {
-        log.snapshot()
-      }
+    addToLogs(log)
 
-      // if end was passed in
-      // go ahead and end
-      if (log.get('end')) {
-        log.end({ silent: true })
-      }
+    if (options.sessionInfo) {
+      Cypress.emit('session:add', log.toJSON())
+    }
 
-      if (log.get('error')) {
-        log.error(log.get('error'), { silent: true })
-      }
+    if (options.emitOnly) {
+      return
+    }
 
-      log.wrapConsoleProps()
+    triggerLog(log)
 
-      const onBeforeLog = state('onBeforeLog')
+    // if not current state then the log is being run
+    // with no command reference, so just end the log
+    if (!current) {
+      log.end()
+    }
 
-      // dont trigger log if this function
-      // explicitly returns false
-      if (_.isFunction(onBeforeLog)) {
-        if (onBeforeLog.call(cy, log) === false) {
-          return
-        }
-      }
+    return log
+  }
 
-      // set the log on the command
-      const current = state('current')
+  logFn._logs = logs
 
-      if (current) {
-        current.log(log)
-      }
+  return logFn
+}
 
-      addToLogs(log)
+export default {
```

Comment:
nit: it would be cool to get rid of `export default { function1, function2 }` and use `export function1; export function2` where possible

Reply:
I didn't change it because we're currently importing them as `import $Log from './log'`. I guess this can be fixed after refactoring log to class.